docs/reference/remap.cue

```diff
@@ -65,4 +64,205 @@ remap: {
+				example:
+
+				```json
+				{"status_code":200,"username":"booper1234","message":"Successful transaction"}
+				```
+
+				In this case, the event, represented by the dot, has three fields: `.status_code`,
+				`.username`, and `.message`. You can assign new values to the existing fields
+				(`.message = "something different"`), add new fields (`.new_field = "new value"`),
+				delete fields (`del(.username)`), store those values in variables (`$code =
+				.status_code`), and more.
+
+				The dot can also represent nested values, for example `.transaction.id` or
+				`.geo.latitude`.
+				"""
+			examples: [
+				".",
+				".status_code",
+				".message",
+				".username",
+				".transaction.id",
+				".geo.latitude",
+			]
+		}
+
+		"Functions": {
+			href: "functions"
+
+			description: """
+				In TRL, functions act just like they do in standard programming languages. They can
+				take inputs (or no input) and either return values or make an
+				[assertion](#assertion) using the [`assert`](#assert) function.
+
+				TRL types that can serve as inputs to functions are listed in [Parameter
+				types](#parameter-types); types that functions can output are listed in [Return
+				types](#return-types).
+				"""
+
+			examples: [
+				"parse_json(.message)",
+				"assert(.status_code == 500)",
+				#"ip_subnet(.address, "255.255.255.0")"#,
+			]
+		}
+
+		"Assignment": {
+			href: "assignment"
+
+			description: """
+				You can assign values to fields using a single equals sign (`=`). If the field
+				already exists, its value is re-assigned; it the field doesn't already exist, it's
+				created and assigned the value.
+				"""
+
+			examples: [
+				".request_id = uuidv4()",
+				".average = .total / .number",
+				".partition_id = .status_code",
+				".is_server_error = .status_code == 500",
+			]
+		}
+
+		"Assertion": {
+			href: "assertion"
+
+			description: """
+				In TRL, you can assert that something is true using [`assert`](#assert). If an
+				assertion succeeds, Vector does nothing and continues with the script; if it fails,
+				the script stops executing and an error is logged. `assert` can take any Boolean as
+				input, including functions that return Booleans, like [`exist`](#exist).
+				"""
+
+			examples: [
+				#"assert(.level == "warn")"#,
+				"assert(exists(.user.username))",
+				"assert(10 == 10 && 1.0 == 1.0)",
+				#"assert(ip_cidr_contains(.address, "192.168.0.0/16"))"#,
+				#"assert(contains("Vector rocks", "vector", case_sensitive = false))"#,
+			]
+		}
+
+		"Operators": {
+			href: "operators"
+
+			description: """
+				TRL offers a standard set of operators that should be familiar from many other
+				programming languages.
+				"""
+
+			examples: [
+				"exists(.request_id) && !exists(.username)",
+				".status_code == 200",
+				#".user.plan != "enterprise" && .user.role == "admin""#,
+			]
+
+			operators: {
+				"Boolean": {
+					"&&": "And"
+					"||": "Or"
+					"!":  "Not"
+				}
```

Comment:
Does it make more sense to invert this structure? With the operators at the top and the category as a key? I'm fine either way, it just jumped out to me.

Reply:
The logical grouping seems more intuitive to me but I'm not dead set on it